### ISSUE TITLE
BUG: train_scorer.py runs matplotlib visualization code outside __main__ guard causing import side effects

### ISSUE THREAD

**Issue by shreyjeetdas** [ronisarkarexe/ai-stories-book#879]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What happened?

At the bottom of backend/ml/train_scorer.py, the matplotlib evaluation and plotting code runs at module level instead of being placed inside the if __name__ == "__main__": block.

This means the following code executes automatically whenever the module is imported:

pred_coh, pred_cre, pred_rel = model.predict(...)
fig, axes = plt.subplots(1, 3, figsize=(15, 5))
...
plt.savefig("scorer_eval.png")

If another module imports from train_scorer.py, the evaluation code runs immediately and may call model.predict() before the model is initialized, causing crashes or runtime errors.

Expected behavior:

Evaluation and plotting code should only execute when the script is run directly.

Actual behavior:

Evaluation and plotting code executes on import.

### Steps to reproduce

Open backend/ml/train_scorer.py
Observe that the matplotlib evaluation code is outside the if __name__ == "__main__": block
Create another Python file and import the module:
from backend.ml.train_scorer import *
Run the importing file
Notice that the plotting/evaluation logic executes automatically during import
If the model is not initialized, model.predict() may fail

### Environment (if relevant)

Python 3.x
Matplotlib
TensorFlow / Keras model pipeline
Backend ML training module

### Screenshots or screen recording (if helpful)

N/A

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

**Comment by shreyjeetdas** [comment]
I meant label it as GSSoC'26
